google/cloud/storage/internal/xml_node.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_XML_NODE_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_XML_NODE_H
+
+#include "google/cloud/storage/version.h"
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/**
+ * Represents an XML node in an XML tree.
+ *
+ * @note This is used in the implementation of [GCS MPU]
+ *
+ * [GCS MPU]: https://cloud.google.com/storage/docs/multipart-uploads
+ *
+ * Normally a single node represents an XML element(tag), but we also
+ * treat a text portion as a node. If the tag_name_ is empty, it's considered as
+ * a text node.
```

Comment:
Nit: Reflow the line breaks. Something appears to be either too short or too long.

Reply:
Thanks, done